containers/ome-tiff-offsets/context/main.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+from glob import glob
+from pathlib import Path
+from os import mkdir
+
+from aicsimageio import AICSImage
+from aicsimageio.writers import ome_tiff_writer
+from tifffile import TiffFile
+
+def get_offsets(tiff_filepath):
+    with TiffFile(tiff_filepath) as tif:
+        offsets = [page.offset for page in tif.pages]
+    return offsets
+
+
+
+def main(input_dir, output_dir):
+    try:
+        mkdir(output_dir)
+    except FileExistsError:
+        pass
+    # potentially we could have .ome.tiff
+    for input in glob(input_dir + '/*.ome.tif*'):
+        # get image metadata and image data
+        with AICSImage(input) as input_image:
+            image_data_from_input = input_image.get_image_data()[0]
+            omexml = input_image.metadata
+        # get offsets and add them to the omexml as structured annotations
+        offsets = get_offsets(input)
+        structured_annotations = omexml.structured_annotations
+        structured_annotations.add_original_metadata(key='IFD_Offsets', value=str(offsets))
+        # create the new output path for the ome tiff
+        input_path = Path(input)
+        new_ome_tiff_path = Path(output_dir) / input_path.name
+        # write the file out
+        with open(Path(output_dir) / 'ome.xml', 'w') as xml_write:
+            xml_write.write(str(omexml))
```

Comment:
I'm not sure what kind of object omexml is, but is there way to pretty-print it, so a line oriented diff would be useful, if that's ever necessary?

Reply:
or if the object itself doesn't offer this, wrap it in an xml object that does.